src/test/java/org/apache/sysml/test/integration/functions/codegenalg/AlgorithmAutoEncoder.java

```diff
@@ -120,50 +128,54 @@ public void testAutoEncoder512SparseRewritesSpark() {
 	
 	private void runGLMTest(int batchsize, boolean sparse, boolean rewrites, ExecType instType)
 	{
-		boolean oldFlag = OptimizerUtils.ALLOW_ALGEBRAIC_SIMPLIFICATION;
-		RUNTIME_PLATFORM platformOld = rtplatform;
-		switch( instType ){
-			case MR: rtplatform = RUNTIME_PLATFORM.HADOOP; break;
-			case SPARK: rtplatform = RUNTIME_PLATFORM.SPARK; break;
-			default: rtplatform = RUNTIME_PLATFORM.HYBRID_SPARK; break;
-		}
-	
-		boolean sparkConfigOld = DMLScript.USE_LOCAL_SPARK_CONFIG;
-		if( rtplatform == RUNTIME_PLATFORM.SPARK || rtplatform == RUNTIME_PLATFORM.HYBRID_SPARK )
-			DMLScript.USE_LOCAL_SPARK_CONFIG = true;
+		for(TestType testType:TestType.values()) {
+			currentTestType = testType;
```

Comment:
Similarly for all the following tests, it would be good to leave the core test as they are, add the testType as a parameter to runGLMTest and similar functions and explicitly add tests. I would be good not to replicate all tests (in the interest of test time) but maybe just add a subset of new tests for the heuristics (e.g., just one batchsize and with rewrites, but for both dense/sparse and CP/SPARK).